### PR TITLE
Add reference implementation for computing RESP charges

### PR DESCRIPTION
**Pull request by alenaizan**
## Description
Reference implementation for computing restrained electrostatic potential charges.

### CODE REVIEW

File: Restrained-Electrostatic-Potential/espfit.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    if difm > toler:
+        note += '\nCharge fitting did not converge; ' +\
+               'try increasing the maximum number of iterations to ' +\
+               '> %i.' %maxit
+    return q_q, note
+
+def intramolecular_constraints(constraint_charge, constraint_equal, constraint_groups):
+    """Function to extract intramolecular constraints from user constraint input
+       input:
+             constraint_charge: list of lists of charges and atom indecies list
```
Comment (dgasmith) [inline Restrained-Electrostatic-Potential/espfit.py]: Can you use [numpy style docs](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) for the whole repository?

**Comment by alenaizan** [comment]
I fixed the documentation. For the name I don't think we should change it because what I am doing here in the code is mostly the charge fitting part, not the computation of the quantum electrostatic potential. Or maybe this can be a sub-directory inside a One-Electron-Property

**Comment by dgasmith** [comment]
Please do move this to a subdirectory of the base tree. The directory can be called ` One-Electron-Property`; however, if you come up with a better name that is relatively generic please use that.

**Comment by alenaizan** [comment]
fixed.

File: One-Electron-Property/Restrained-Electrostatic-Potential/espfit.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
+import numpy as np
```
Comment (dsirianni) [inline One-Electron-Property/Restrained-Electrostatic-Potential/espfit.py]: For each of these scripts, could you please add appropriate headers according to the conventions [here](https://github.com/psi4/psi4numpy/blob/master/CONTRIBUTING.md#header-styleguide)?

**Comment by dgasmith** [comment]
@alenaizan @dsirianni Can you work together to get this in? Important to do so in the next several days as the paper reviews are back.

**Comment by alenaizan** [comment]
Sure, I will do it over the weekend.

**Comment by alenaizan** [comment]
Changed according to the suggestions.

**Comment by dgasmith** [comment]
@dsirianni Please re-review this as expediently as you can as you have a blocking review.

**Pull request merged**